app/main.py

```diff
@@ -1,2 +1,75 @@
+from fractions import Fraction
+from typing import Any, Hashable
+
+
 class Dictionary:
-    pass
+    class Node:
+        def __init__(self, key: Hashable, value: Any) -> None:
+            self.key = key
+            self.value = value
```

Comment:
would make sense to keep hash from key here as well and check if we have a match in later functions